Lib9c/BlockPolicySource.cs

```diff
@@ -103,5 +108,29 @@ private void UpdateActivationSet()
                 UpdateActivationSet(state);
             }
         }
+
+        private InvalidBlockException ValidateBlock(Block<NCAction> block)
+        {
+            if (AuthorizedMinersState is null)
+            {
+                return null;
+            }
+
+            bool targetBlock =
+                0 < block.Index && block.Index <= AuthorizedMinersState.ValidUntil &&
+                block.Index % AuthorizedMinersState.Interval == 0 && 
+                block.Miner is Address miner;
+            bool minedByAuthorities = AuthorizedMinersState.Miners.Contains(miner);
+            
+            if (targetBlock && !minedByAuthorities)
```

Comment:
Due to short-circuit evaluation, there seems a case that `miner` is uninitialized.  Though the whole logic would work as intended whether `miner` is initialized or not, it's quite complicated at first glance.  It would be better if the whole condition is rewritten into nested `if` blocks IMHO.